WordPress/Classes/Models/Blog+History.swift

```diff
@@ -0,0 +1,42 @@
+import Foundation
+
+extension Blog {
+
+    /// Returns the blog currently flagged as the one last used, or the primary blog,
+    /// or the first blog in an alphanumerically sorted list, whichever is found first.
+    @objc(lastUsedOrFirstBlogInContext:)
+    static func lastUsedOrFirstBlog(in context: NSManagedObjectContext) -> Blog? {
+        lastUsedBlog(in: context)
+            ?? (try? WPAccount.lookupDefaultWordPressComAccount(in: context))?.defaultBlog
+            ?? firstBlog(in: context)
+    }
+
+    /// Returns the blog currently flaged as the one last used.
+    static func lastUsedBlog(in context: NSManagedObjectContext) -> Blog? {
+        guard let url = RecentSitesService().recentSites.first else {
+            return nil
+        }
+
+        return blog(with: NSPredicate(format: "visible = YES AND url = %@", url), in: context)
+    }
+
+    private static func firstBlog(in context: NSManagedObjectContext) -> Blog? {
+        blog(with: NSPredicate(format: "visible = YES"), in: context)
+    }
+
+    private static func blog(with predicate: NSPredicate, in context: NSManagedObjectContext) -> Blog? {
```

Comment:
This function is translated from [`BlogService.blogsWithPredicate`](https://github.com/wordpress-mobile/WordPress-iOS/blob/e465e5514e3467afe972171a987fd5607e0cfa0d/WordPress/Classes/Services/BlogService.m#L832-L848)